src/Maestro/Maestro.Web/Api/v2019_01_16/Models/Goal.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Maestro.Data.Migrations;
+using Maestro.Data.Models;
+
+namespace Maestro.Web.Api.v2019_01_16.Models
+{
+    public class Goal
+    {
+        public Goal([NotNull] Data.Models.GoalTime other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentNullException(nameof(other));
+            }
+            DefinitionId = other.DefinitionId;
+            Minutes = other.Minutes;
+            ChannelId = other.ChannelId;
+        }
+
+        public int DefinitionId { get; set; }
+        public int ChannelId { get; set; }
```

Comment:
Both method take the "ChannelName", the string.  It would be nicer if we were consistent. Either taking and returning the name, or taking and returning the number.

Reply:
So we are taking in ChannelName But while storing the info about Channel, we just store the id, as it is a foreign key to connect to Channel table. 

For the get request also we are taking in Channel Name. So for both Put and Get we use channelName.

Reply:
I had a similar concern, Chad, so I looked around at the other darc commands, and all of them are like this: they take a channel name (because we can't expect our customers to memorize all of the id numbers that are in the database), and store the channel id because it's the foreign key. It seems weird and hard to deal with, but it is consistent with the rest of the code base.

Reply:
The issue is that this isn't the database model, this is the model that customers see.

Reply:
Ah, didn't look where this was. The other models use

```
public v2018_07_16.Models.Channel Channel { get; }
```

Reply:
sending back Channel obj instead of Channel id.